crates/autopilot/src/run_loop.rs

```diff
@@ -405,20 +408,55 @@ impl RunLoop {
 
     /// Execute the solver's solution. Returns Ok when the corresponding
     /// transaction has been mined.
-    async fn settle(&self, driver: &infra::Driver, solved: &Solution) -> Result<(), SettleError> {
+    async fn settle(
+        &self,
+        driver: infra::Driver,
+        solved: &Solution,
+        auction_id: i64,
+    ) -> Result<(), SettleError> {
         let order_ids = solved.order_ids().copied().collect();
         self.persistence
             .store_order_events(order_ids, OrderEventLabel::Executing);
 
+        let eth = self.eth.clone();
+        let persistence = self.persistence.clone();
+        let (cancellation_tx, cancellation_rx) = watch::channel(false);
+        let settlement_tx_wait_task = tokio::spawn(async move {
+            let tag = auction_id.to_be_bytes();
+            Self::wait_for_settlement_transaction(eth, persistence, &tag, cancellation_rx)
+                .await
```

Comment:
This execution might also be wrapped with an additional timeout(in case there are on-chain issues).

Reply:
Wrapping that with timeout is definitely a good idea since the function has a potential infinite `loop`

Reply:
Currently, the task is cancelled when the driver settlement task finishes the execution, which has a timeout configured. So with the current setup, the second timeout would be redundant.